src/emulator/module/tests/arg_layout_tests.cpp

```diff
@@ -40,7 +40,7 @@ static std::ostream &operator<<(std::ostream &out, const ArgLayout &layout) {
 }
 
 TEST(lay_out, gpr_overflows_to_stack) {
-    const auto actual = lay_out<int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t>();
+    const auto actual = std::get<0>(lay_out<int8_t, uint8_t, int16_t, uint16_t, int32_t, uint32_t>());
```

Comment:
I think I would have preferred that `expected` was fixed to be the right type, then the unit tests can test that the final states are correct.

Reply:
Please also add a unit test for a typical printf style function, so we check that the vararg does not affect the arg layout. i.e., the format arg is added but the vararg does not take up any extra register or stack space.

Reply:
The varargs does indeeds take extra space as you say. I will remove unneccessary codes (in the lay_out_args)